File: SolvedProblems/CountGoodNodesInBinaryTree/CountGoodNodesInBinaryTree.py
def goodNodes( root):
    queue = [root]
    maxQueue = [-float("inf")]
    count = 0

    while (len(queue)):
      currNode = queue.pop()
      currMax = maxQueue.pop()

      currMax = max(currNode.val, currMax)

      if (currMax == currNode.val):
          count = count + 1

      if (currNode.left):
          queue.append(currNode.left)
          maxQueue.append(currMax)

      if (currNode.right):
          queue.append(currNode.right)
          maxQueue.append(currMax)


    return count
        

File: SolvedProblems/CountGoodNodesInBinaryTree/CountGoodNodesInBinaryTree.js
const goodNodes = (root) => {
  if (!root.left && !root.right) return 1;

  let queue = [root];
  let maxQueue = [-Infinity];
  let count = 0;

  while (queue.length) {
    let currNode = queue.pop();
    let max = maxQueue.pop();
    // there is a diff max in every floor
    max = Math.max(currNode.val, max);

    if (max === currNode.val) count++;

    if (currNode.left) {
      queue.push(currNode.left);
      maxQueue.push(max);
    }

    if (currNode.right) {
      queue.push(currNode.right);
      maxQueue.push(max);
    }
  }

  return count;
};
